File: lib/host/break.c
/* 
* Copyright (C) 2009,2010 Toni Gundogdu.
*
* This program is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

#include "host.h"

_host_constants(break, "break.com", "flv");

_host_re(re_id,     "(?i)contentid='(.*?)'");
_host_re(re_title,  "(?i)id=\"vid_title\" content=\"(.*?)\"");
_host_re(re_fpath,  "(?i)contentfilepath='(.*?)'");
_host_re(re_fname,  "(?i)filename='(.*?)'");
_host_re(re_fhash,  "(?i)flashvars.icon = \"(.*?)\"");

QUVIcode
handle_break(const char *url, _quvi_video_t video) {
    char *content, *fpath, *fname, *fhash;
    QUVIcode rc;

    /* host id */
    _host_id("break");

    /* common */
    rc = parse_page_common(url, video, &content, re_id, re_title);

    if (rc != QUVI_OK)
        return (rc);

    /* fpath */
    rc = regexp_capture(
        video->quvi,
        content,
        re_fpath,
        0,
        0,
        &fpath,
        (void *) 0
    );

    if (rc != QUVI_OK) {
        _free(content);
        return (rc);
    }

    /* fname */
    rc = regexp_capture(
        video->quvi,
        content,
        re_fname,
        0,
        0,
        &fname,
        (void *) 0
    );

    if (rc != QUVI_OK) {
        _free(content);
        _free(fpath);
        return (rc);
    }

    /* fhash */
    rc = regexp_capture(
        video->quvi,
        content,
        re_fhash,
        0,
        0,
        &fhash,
        (void *) 0
    );

    _free(content);

    if (rc != QUVI_OK) {
        _free(fpath);
        _free(fname);
        return (rc);
    }

    /* video link */
    add_video_link(&video->link,
        "http://video1.break.com/dnet/media/%s/%s.flv?%s",
            fpath, fname, fhash);

    _free(fpath);
    _free(fname);
    _free(fhash);

    return (QUVI_OK);
}

File: share/lua/website/break.lua

-- quvi
-- Copyright (C) 2010  Toni Gundogdu <legatvs@gmail.com>
--
-- This file is part of quvi <http://quvi.sourceforge.net/>.
--
-- This library is free software; you can redistribute it and/or
-- modify it under the terms of the GNU Lesser General Public
-- License as published by the Free Software Foundation; either
-- version 2.1 of the License, or (at your option) any later version.
--
-- This library is distributed in the hope that it will be useful,
-- but WITHOUT ANY WARRANTY; without even the implied warranty of
-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-- Lesser General Public License for more details.
--
-- You should have received a copy of the GNU Lesser General Public
-- License along with this library; if not, write to the Free Software
-- Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
-- 02110-1301  USA
--

-- Identify the script.
function ident (page_url)
    local t   = {}
    t.domain  = "break.com"
    t.formats = "default"
    t.handles = (page_url ~= nil and page_url:find(t.domain) ~= nil)
    return t
end

-- Parse video URL.
function parse (video)
    video.host_id = "break"
    local page    = quvi.fetch(video.page_url)

    local _,_,s = page:find('id="vid_title" content="(.-)"')
    video.title = s or error ("no match: video title")

    local _,_,s = page:find("ContentID='(.-)'")
    video.id    = s or error ("no match: video id")

    local _,_,s = page:find("FileName='(.-)'")
    local fname = s or error ("no match: content file name")

    local _,_,s = page:find('flashVars.icon = "(.-)"')
    local fhash = s or error ("no match: file hash")

    video.url = {
        string.format("%s.flv?%s", fname,fhash)
    }

    return video
end

